cypress/pages/BounceRulesContainer/index.js

```diff
@@ -82,42 +79,82 @@ class BounceRulesPage extends Page {
     return cy.get(Selectors.confirmationSubmit);
   }
 
-  get testRule() {
-    return cy.get("[data-rule='cypressTest']");
+  get createdDeleteRuleAPI() {
+    return cy.get('[data-cypress="cypressDeleteTest"]');
   }
 
-  fillForm(
-    priority,
-    bounceAction,
-    responseCode,
-    description,
-    enhancedCode,
-    regex
-  ) {
-    if (priority) {
-      this.priority.type(priority);
-    }
-
-    if (bounceAction) {
-      this.bounceAction.type(bounceAction);
-    }
-
-    if (responseCode) {
-      this.responseCode.type(responseCode);
-    }
-
-    if (description) {
-      this.description.type(description);
-    }
-    if (enhancedCode) {
-      this.enhancedCode.type(enhancedCode);
-    }
+  get createdCreateRuleAPI() {
+    return cy.get('[data-cypress="cypressCreateTest"]');
+  }
 
-    if (regex) {
-      this.regex.type(regex);
-    }
+  open() {
+    super.open("/bounce_rules");
+  }
 
-    return this.submitButton.click();
+  createdRuleButton(id) {
+    return cy.get(`[data-rule="${id}"]`);
+  }
+
+  deleteBounceRuleAPI(testRule) {
+    return new Promise(resolve => {
+      cy.task("getRules").then(res => {
```

Comment:
cy.task is already a Promise-like thing so you can do things like return cy.task().then() and chain another .then after that. The whole cy.reload().then() => resolve() looks a little hacky. We can avoid having it be too nested by always doing return on one of the cy commands and then doing another .then.